### DISCUSSION TITLE
Can't save battery in game on MacOS

### DISCUSSION THREAD

**Discussion by ElfenSterben** [visualboyadvance-m/visualboyadvance-m#1149] [General]
when i save the game
it raise this error
00:56:04: Error creating file 
00:56:04: Error writing battery 

![](https://images4.imagebam.com/a1/5a/ef/ME1RQAJ_o.png)

### ROM details

![](https://thumbs4.imagebam.com/ec/d7/ce/ME1RQH1_t.png)

**Comment by Squall-Leonhart** [comment]
use the osx permission manager to give it permission to use the file system

https://www.wikihow.com/Give-System-Permissions-for-Apps-on-MacOS-Catalina

**Comment by ElfenSterben** [comment]
> use the osx permission manager to give it permission to use the file system
> 
> https://www.wikihow.com/Give-System-Permissions-for-Apps-on-MacOS-Catalina

i try it 
but it doesn't work,
i can't add it to  "Files and Folders" or "Full Disk Access"

but when i open it in iterm2 with sudo, it works

`
sudo open VBA/visualboyadvance-m.app
`

the system version is  macOS Big Sur 11.4

**Comment by ElfenSterben** [comment]
> use the osx permission manager to give it permission to use the file system
> 
> https://www.wikihow.com/Give-System-Permissions-for-Apps-on-MacOS-Catalina

i make a record 

https://www.youtube.com/watch?v=vX5MFILf2xQ

**Comment by jhiajhainjhon** [comment]
signed up an account just to join this conversation, tried a few things not sure which makes it works (macOS Big Sur 11.3.1)
here's what I did below
1. create the ''visualboyadvance-m'' folder in /Applications, and create ''Battery Saves'',''Save States'',''Screenshots'' folders in it
2. right-click ''Battery Saves'',''Save States'',''Screenshots'' folders and Enable their Folder Actions (don't add any script)
3. right-click ''visualboyadvance-m.app'' to make it ''Locked'' in the info
4. place all .gba game roms in the ''visualboyadvance-m'' folder and rename them in english
<img width="1032" alt="圖片" src="https://user-images.githubusercontent.com/87281918/125210213-d506ce00-e2d0-11eb-98a1-0e8df28612b3.png">
5. open the ''visualboyadvance-m.app'' and click options-->Directories to set them like this
<img width="954" alt="圖片" src="https://user-images.githubusercontent.com/87281918/125209742-902d6800-e2cd-11eb-8a7d-6f7863d5d006.png">
AND IT JUST WORKS ヽ( ﾟ∀ﾟ)ﾉ
<img width="1032" alt="圖片" src="https://user-images.githubusercontent.com/87281918/125210319-a76e5480-e2d1-11eb-8ad1-b676583c8134.png">
<img width="1032" alt="圖片" src="https://user-images.githubusercontent.com/87281918/125210323-ad643580-e2d1-11eb-8d2e-f0d1c4b5486c.png">
hope this works for you too(๑•̀ㅂ•́)و✧

**Comment by ElfenSterben** [comment]
> signed up an account just to join this conversation, tried a few things not sure which makes it works (macOS Big Sur 11.3.1)
> here's what I did below
> 
> 1. create the ''visualboyadvance-m'' folder in /Applications, and create ''Battery Saves'',''Save States'',''Screenshots'' folders in it
> 2. right-click ''Battery Saves'',''Save States'',''Screenshots'' folders and Enable their Folder Actions (don't add any script)
> 3. right-click ''visualboyadvance-m.app'' to make it ''Locked'' in the info
> 4. place all .gba game roms in the ''visualboyadvance-m'' folder and rename them in english
> 
> <img alt="圖片" width="1032" src="https://user-images.githubusercontent.com/87281918/125210213-d506ce00-e2d0-11eb-98a1-0e8df28612b3.png">
> 
> 5. open the ''visualboyadvance-m.app'' and click options-->Directories to set them like this
> <img alt="圖片" width="954" src="https://user-images.githubusercontent.com/87281918/125209742-902d6800-e2cd-11eb-8a7d-6f7863d5d006.png">
> 
> AND IT JUST WORKS ヽ( ﾟ∀ﾟ)ﾉ
> <img alt="圖片" width="1032" src="https://user-images.githubusercontent.com/87281918/125210319-a76e5480-e2d1-11eb-8ad1-b676583c8134.png">
> 
> <img alt="圖片" width="1032" src="https://user-images.githubusercontent.com/87281918/125210323-ad643580-e2d1-11eb-8d2e-f0d1c4b5486c.png">
> 
> hope this works for you too(๑•̀ㅂ•́)و✧

thank you 
it works ~ 
i try it and find that it's only because the .gba filename is Chinese
